File: service/src/main/java/com/ricardococati/service/converter/MediaMovelExponencialConverter.java
package com.ricardococati.service.converter;

import com.ricardococati.model.entities.CandlestickDiario;
import com.ricardococati.model.entities.MediaMovelExponencial;
import com.ricardococati.model.entities.MediaMovelExponencialDiario;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class MediaMovelExponencialConverter {

  public MediaMovelExponencialDiario converterCandlestickDiarioToMediaMovelExponencial(
      CandlestickDiario candlestickDiario){
    return MediaMovelExponencialDiario
        .builder()
        .mediaMovelExponencial(
            MediaMovelExponencial
                .builder()
                .codneg(candlestickDiario.getCandlestick().getCodneg())
                .build())
        .build();
  }
}

File: service/src/main/java/com/ricardococati/service/converter/MediaMovelExponencialConverter.kt
package com.ricardococati.service.converter

import com.ricardococati.model.entities.CandlestickDiario
import com.ricardococati.model.entities.MediaMovelExponencial
import com.ricardococati.model.entities.MediaMovelExponencialDiario
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class MediaMovelExponencialConverter {

    fun converterCandlestickDiarioToMediaMovelExponencial(
            candlestickDiario: CandlestickDiario): MediaMovelExponencialDiario {
        return MediaMovelExponencialDiario
                .builder()
                .mediaMovelExponencial(
                        MediaMovelExponencial
                                .builder()
                                .codneg(candlestickDiario.candlestick!!.codneg)
                                .build())
                .build()
    }

    companion object {
        private val log = LoggerFactory
                .getLogger(MediaMovelExponencialConverter::class.java)
    }

}